### PR TITLE
Enhancing Windows 10 open source build documentation in INSTALL.md

### PR DESCRIPTION
**Pull request by ANS-UXI**
I was trying to configure the repo with open source build in Windows 10 but then I faced a lot of errors and difficulties building and running the scripts. All thanks to @biodranik for helping me out and fix the errors. I documented what he specified and what I did to successfully configure the repo for an open source build in this PR.

### CODE REVIEW

**Review by biodranik — COMMENTED** [review]
The documentation should be updated in place, where it is relevant, instead of adding some new paragraphs.

**Comment by ANS-UXI** [comment]
I made the changes, please have a look!

File: docs/INSTALL.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 To build and run Organic Maps you'll need a machine with at least 4Gb of RAM and 20-30Gb of disk space depending on your target platform. Expect to download 5-10Gb of files.
 
-For _Windows_ you need to have [Git for Windows](https://git-scm.com/download/win) installed and Git bash available in the PATH.
+For _Windows_ you need to have [Git for Windows](https://git-scm.com/download/win) installed and Git bash available in the PATH, make sure you have also installed [Windows 10/11 SDK](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/).
```
Comment (biodranik) [inline docs/INSTALL.md:13]: This line can be moved below, it is needed only for the non-WSL case.

File: docs/INSTALL.md [hunk 2]
```diff
@@ -43,12 +43,12 @@ Configure the repository for an opensource build:
 bash ./configure.sh
 ```
 
-For _Windows 10_: Use WSL to run `./configure.sh`, or, alternatively, run the following command from the
-[Visual Studio Developer Command Prompt](https://docs.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022)
-and follow instructions:
+For _Windows 10_: Use WSL(make sure [g++ is installed](https://askubuntu.com/a/915751)) to run `./configure.sh`, or, alternatively, install the [Visual Studio Developer Command Prompt](https://docs.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022) (make sure to choose the "Desktop development with C++" as your workload while installing Visual Studio), check whether [cl.exe](https://learn.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-170) is in the path by typing `cl` in Developer Command Prompt for VS. Run the following command and follow instructions:
```
Comment (biodranik) [inline docs/INSTALL.md:46]: Let's describe separately two setups:
1. With WSL + installed g++ (sudo apt install g++), no Developer command prompt is needed.
2. Without WSL, but with Visual Studio Installer with MSVC build tools + (likely) Windows SDK (I assume that only Windows 11 may need it, Windows 10 likely will work without it), and with the Developer command prompt.

Comment (ANS-UXI) [reply docs/INSTALL.md]: I have defined two setups, you may review it.

File: docs/INSTALL.md [hunk 3]
```diff
@@ -43,12 +43,21 @@ Configure the repository for an opensource build:
...
 
-For _Windows 10_: Use WSL to run `./configure.sh`, or, alternatively, run the following command from the
-[Visual Studio Developer Command Prompt](https://docs.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022)
-and follow instructions:
+For _Windows 10_:  You should be able to build the project by following either of these setups:
+
+**Setup 1: Using WSL**
+1. Install [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) on your machine.
+2. Install g++ by running the following command in WSL: `sudo apt install g++`
+3. Execute `bash ./configure` in WSL.
```
Comment (biodranik) [inline docs/INSTALL.md:51]:
```suggestion
3. Run `./configure.sh` in WSL.
```

File: docs/INSTALL.md [hunk 4]
```diff
@@ -43,12 +43,21 @@ Configure the repository for an opensource build:
...
+4. After bootstrapping, don't forget to generate header files by `cd 3party/boost` and executing `./b2 headers`.
```
Comment (biodranik) [inline docs/INSTALL.md:52]: This step should not be needed, right? Can you please double-check it, by running
```
git clean -xfd
git submodule foreach --recursive git clean -xfd
```
and then repeating these steps?

File: docs/INSTALL.md [hunk 5]
```diff
@@ -43,12 +43,21 @@ Configure the repository for an opensource build:
...
+
+**Setup 2: Using Visual Studio Developer Command Prompt**
+1. Install the [Visual Studio Developer Command Prompt](https://docs.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022) (make sure to choose the "Desktop development with C++" as your workload while installing Visual Studio).
```
Comment (biodranik) [inline docs/INSTALL.md:55]: Are you sure that all these components are necessary? I assume that MSVC build tools should be enough, no? Would you please try it?

File: docs/INSTALL.md [hunk 6]
```diff
@@ -43,12 +43,21 @@ Configure the repository for an opensource build:
...
+2. Install [Windows 10/11 SDK](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/) if you haven't already.
```
Comment (biodranik) [inline docs/INSTALL.md:56]: Can't SDK be installed from Visual Studio installer?

File: docs/INSTALL.md (hunk 4 above)
Comment (ANS-UXI) [reply docs/INSTALL.md:52]: Ok, now I know step 4 isn't required.

File: docs/INSTALL.md (hunk 5 above)
Comment (ANS-UXI) [reply docs/INSTALL.md:55]: Actually Desktop development with C++ as workload is not needed, one need to just install Visual Studio and MSVC build tools as individual components.

File: docs/INSTALL.md (hunk 6 above)
Comment (ANS-UXI) [reply docs/INSTALL.md:56]: Yes, it can be installed from VS installer, making changes to the PR shortly!

**Comment by ANS-UXI** [comment]
I made the required changes, please have a look.

**Review by biodranik — APPROVED** [review]
Thanks! It looks great now!

**Comment by ANS-UXI** [comment]
> Thanks! It looks great now!

Glad I could contribute! Thanks for guiding me🙏🏻

**Comment by biodranik** [comment]
@ANS-UXI  There is a section in manual that may be updated/removed: https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md#windows-10-manual-boost-library-initialization

**Comment by ANS-UXI** [comment]
> @ANS-UXI There is a section in manual that may be updated/removed: https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md#windows-10-manual-boost-library-initialization

Yeah actually there is no need to manually boost the lib in win10 because it gets done while configuring the repo for an open source build which I added earlier, thanks for informing and I'll remove it!

**Pull request merged**